central/role/datastore/datastore_impl.go

```diff
@@ -287,6 +303,14 @@ func (ds *dataStoreImpl) GetAllAccessScopes(ctx context.Context) ([]*storage.Sim
 	return scopes, nil
 }
 
+func (ds *dataStoreImpl) CountAccessScopes(ctx context.Context) (int, error) {
+	if ok, err := roleSAC.ReadAllowed(ctx); !ok || err != nil {
+		return 0, err
+	}
+
+	return ds.CountAccessScopes(ctx)
```

Comment:
Shouldn't it be `ds.accessScopeStorage.Count(ctx)`?

Reply:
OMG